### PR TITLE
casmuser-3199: Add info for K3s MetalLB pool

### PR DESCRIPTION
**Pull request by keopp-hpe**
#### Summary and Scope
Document the ability to get K3s MetalLB IPs from SLS.

### CODE REVIEW

File: docs/portal/developer-portal/advanced/Enabling_K3s.md [hunk 1]
```diff
@@ -30,8 +30,51 @@ The following steps should be completed prior to configuring the UAN with K3s.
    **Note**: In a future release, additional UANs will be able to join as extra manager or worker nodes. 
 
 1. Identify a pool of IP addresses for the services running in K3s.
+
+   **Note**: The identification of a pool of IP addresses for the services running in K3s should be made at system installation time in order to avoid the possibility of IP collisions.
```
Comment (darrenn-jackson) [inline docs/portal/developer-portal/advanced/Enabling_K3s.md:34]: We need this advice somewhere in the UAN install procedures.

File: docs/portal/developer-portal/advanced/Enabling_K3s.md [hunk 2]
```diff
@@ -30,8 +30,51 @@ The following steps should be completed prior to configuring the UAN with K3s.
...
    
-   This address pool must be routable from the UAN control-plane, and should be unused for other purposes.
+   This address pool must be routable from the UAN control-plane, and should be unused for other purposes. It is suggested to split the `[CAN|CHN] Dynamic MetalLB` subnet into two subnets, `[CAN|CHN] Dynamic MetalLB` and `[CAN}CHN] Dynamic MetalLB K3s` as shown below. These subnet definitions get uploaded to SLS at installation time by CSI.  Alternatively, the starting and ending IPs for K3s may be defined in `vars/uan_helm.yml` as shown below.
```
Comment (darrenn-jackson) [inline docs/portal/developer-portal/advanced/Enabling_K3s.md:36]:
```suggestion
   This address pool must be routable from the UAN control-plane, and should be unused for other purposes. It is suggested to split the `[CAN|CHN] Dynamic MetalLB` subnet into two subnets, `[CAN|CHN] Dynamic MetalLB` and `[CAN|CHN] Dynamic MetalLB K3s` as shown below. These subnet definitions get uploaded to SLS at installation time by CSI.  Alternatively, the starting and ending IPs for K3s may be defined in `vars/uan_helm.yml` as shown below.
```

**Review by darrenn-jackson — CHANGES_REQUESTED** [review]
two changes.

File: docs/portal/developer-portal/advanced/Enabling_K3s.md (hunk 1 above)
Comment (keopp-hpe) [reply docs/portal/developer-portal/advanced/Enabling_K3s.md:34]: Actually, it needs to be in the install of the system docs, likely CSM, so the subnets are defined before CSM is installed.  If an admin/installer waits until after CSM is started, there is a risk of the upper half of the [CAN|CHN] Dynamic MetalLB subnet being used by CSM services which would end up with IP collisions for UAN K3s services.

Comment (keopp-hpe) [reply docs/portal/developer-portal/advanced/Enabling_K3s.md:34]: I added the info to the uan installation prereqs doc.

**Review by darrenn-jackson — APPROVED** [review]

File: docs/portal/developer-portal/advanced/Enabling_K3s.md [hunk 3]
```diff
@@ -30,8 +30,51 @@ The following steps should be completed prior to configuring the UAN with K3s.
...
+            "VlanID": 6,
+
+    ### New [CAN|CHN] Dynamic MetalLB K3s
+            "CIDR": "10.103.7.224/27",
+            "FullName": "CAN Dynamic MetalLB K3s",
+            "Gateway": "10.103.7.225",
+            "MetalLBPoolName": "customer-access-k3s",
+            "Name": "can_metallb_k3s_address_pool",
+            "VlanID": 6,
+    ```
```
Comment (alanm-hpe) [inline docs/portal/developer-portal/advanced/Enabling_K3s.md:65]: Please obfuscate IP addresses with x.x.x.x, y.y.y.y, etc..

File: docs/portal/developer-portal/advanced/Enabling_K3s.md [hunk 4]
```diff
@@ -30,8 +30,51 @@ The following steps should be completed prior to configuring the UAN with K3s.
...
 
 1. Identify a pool of IP addresses for the services running in K3s.
+
+   **Note**: The identification of a pool of IP addresses for the services running in K3s should be made at system installation time in order to avoid the possibility of IP collisions.
    
-   This address pool must be routable from the UAN control-plane, and should be unused for other purposes.
+   This address pool must be routable from the UAN control-plane, and should be unused for other purposes. It is suggested to split the `[CAN|CHN] Dynamic MetalLB` subnet into two subnets, `[CAN|CHN] Dynamic MetalLB` and `[CAN|CHN] Dynamic MetalLB K3s` as shown below. These subnet definitions get uploaded to SLS at installation time by CSI.  Alternatively, the starting and ending IPs for K3s may be defined in `vars/uan_helm.yml` as shown below.
+
+    Example of splitting the existing `[CAN|CHN] Dynamic MetalLB` subnet. By default, the `FullName` of the subnet used for K3s is `[CAN|CHN] Dynamic MetalLB K3s`, depending on whether `CAN` or `CHN` is being used for the customer access network.  This default `FullName` may be overridden by setting the `sls_can_metallb_fullname` variable in CFS to the expected name.
```
Comment (alanm-hpe) [inline docs/portal/developer-portal/advanced/Enabling_K3s.md:38]:
```suggestion
    Here is an example of splitting the existing `[CAN|CHN] Dynamic MetalLB` subnet. By default, the `FullName` of the subnet used for K3s is `[CAN|CHN] Dynamic MetalLB K3s`, depending on whether `CAN` or `CHN` is being used for the customer access network.  This default `FullName` may be overridden by setting the `sls_can_metallb_fullname` variable in CFS to the expected name.
```

**Review by darrenn-jackson — APPROVED** [review]

**Review by darrenn-jackson — APPROVED** [review]

**Pull request merged**